app/src/main/java/org/thoughtcrime/securesms/payments/MobileCoinTestNetConfig.java

```diff
@@ -0,0 +1,95 @@
+                                + "uEl0ZmwAI30ZmzgcWc7RBeWD1/zNt55zzhfPRLx/DfDY5Kdp6oFHWMvI2r1/oZkd\n"
+                                + "hjFp7pV6qrl7vOyR5QqmuRkCAwEAAQ==");
+  }
+
+  @Override
+  @NonNull AuthCredentials getAuth() throws IOException {
+    return signalServiceAccountManager.getPaymentsAuthorization();
+  }
+
+  @Override
+  @NonNull ClientConfig getConfig() {
+    try {
+      byte[]               mrEnclaveConsensus  = Hex.toByteArray("9659ea738275b3999bf1700398b60281be03af5cb399738a89b49ea2496595af");
+      byte[]               mrEnclaveReport     = Hex.toByteArray("a4764346f91979b4906d4ce26102228efe3aba39216dec1e7d22e6b06f919f11");
+      byte[]               mrEnclaveLedger     = Hex.toByteArray("768f7bea6171fb83d775ee8485e4b5fcebf5f664ca7e8b9ceef9c7c21e9d9bf3");
+      byte[]               mrEnclaveView       = Hex.toByteArray("e154f108c7758b5aa7161c3824c176f0c20f63012463bf3cc5651e678f02fb9e");
+      byte[]               mrSigner            = Hex.toByteArray("bf7fa957a6a94acb588851bc8767e0ca57706c79f4fc2aa6bcb993012c3c386c");
+      Set<X509Certificate> trustRoots          = getTrustRoots(R.raw.signal_mobilecoin_authority);
+      ClientConfig         config              = new ClientConfig();
+      String[]             hardeningAdvisories = {"INTEL-SA-00334"};
+
+      config.logAdapter = new MobileCoinLogAdapter();
+      config.fogView    = new ClientConfig.Service().withTrustRoots(trustRoots)
+                                                    .withVerifier(new Verifier().withMrEnclave(mrEnclaveView, null, hardeningAdvisories)
+                                                                                .withMrSigner(mrSigner, FOG_VIEW_PRODUCT_ID, SECURITY_VERSION, null, hardeningAdvisories));
+      config.fogLedger  = new ClientConfig.Service().withTrustRoots(trustRoots)
+                                                    .withVerifier(new Verifier().withMrEnclave(mrEnclaveLedger, null, hardeningAdvisories)
+                                                                                .withMrSigner(mrSigner, FOG_LEDGER_PRODUCT_ID, SECURITY_VERSION, null, hardeningAdvisories));
+      config.consensus  = new ClientConfig.Service().withTrustRoots(trustRoots)
+                                                    .withVerifier(new Verifier().withMrEnclave(mrEnclaveConsensus, null, hardeningAdvisories)
+                                                                                .withMrSigner(mrSigner, CONSENSUS_PRODUCT_ID, SECURITY_VERSION, null, hardeningAdvisories));
+      config.report     = new ClientConfig.Service().withVerifier(new Verifier().withMrEnclave(mrEnclaveReport, null, hardeningAdvisories)
+                                                                                .withMrSigner(mrSigner, FOG_REPORT_PRODUCT_ID, SECURITY_VERSION, null, hardeningAdvisories));
+      return config;
+    } catch (AttestationException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)